packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -483,31 +505,112 @@ export class PerformanceEvent {
+ */
+function getValidTelemetryProps(obj: any): ITaggableTelemetryProperties {
+    const props: ITaggableTelemetryProperties = {};
+    for (const key of Object.keys(obj)) {
+        const val = obj[key];
+        switch (typeof val) {
+            case "string":
+            case "number":
+            case "boolean":
+            case "undefined":
+                props[key] = val;
+                break;
+            default: {
+                if (isTaggedTelemetryPropertyValue(val)) {
+                    props[key] = val;
+                }
+                break;
+            }
+        }
+    }
+    return props;
+}
+
 /**
- * - Helper class for error tracking that can be used to log an error in telemetry.
- * - Care needs to be taken not to log PII information!
- * - This allows additional properties to be logged because object of this instance will record all of their properties
- *   when logged with a logger.
- * - Logger ignores all properties from any other error objects (not being instance of LoggingError), with exception of
- *   'message' & 'stack' properties if they exists on error object.
- * - In other words, logger logs only what it knows about and has good confidence it does not container PII information.
+ * Helper class for error tracking that can be used to log an error in telemetry.
+ * The props passed in (and any set directly on the object after the fact) will be
+ * logged in accordance with the given TelemetryDataTag, if present.
+ *
+ * PLEASE take care to properly tag properties set on this object
  */
-export class LoggingError extends Error {
+export class LoggingError extends Error implements ILoggingError {
     constructor(
         message: string,
-        props?: ITelemetryProperties,
+        props?: ITaggableTelemetryProperties,
     ) {
         super(message);
+        this.addTelemetryProperties(props);
+    }
+
+    /**
+     * Add additional properties to be logged
+     */
+    public addTelemetryProperties(props?: ITaggableTelemetryProperties) {
         Object.assign(this, props);
     }
 
-    // Return all properties
-    public getTelemetryProperties(): ITelemetryProperties {
-        const props: ITelemetryProperties = {};
-        // Could not use {...this} because it does not return properties of base class.
-        for (const key of Object.getOwnPropertyNames(this)) {
-            props[key] = this[key];
-        }
-        return props;
+    /**
+     * Get all properties fit to be logged to telemetry for this error
+     */
+    public getTelemetryProperties(): ITaggableTelemetryProperties {
+        const taggableProps = getValidTelemetryProps(this);
+        // Include non-enumerable props inherited from Error that would not be returned by getValidTelemetryProps
+        // But if any were overwritten (e.g. with a tagged property), then use the result from getValidTelemetryProps
+        return  {
+            stack: this.stack,
```

Comment:
stack should have tag = TelemetryDataTag.PackageData (or renamed to be more inclusive) as it's same bucket - it has data from code, and it might not be possible to collect it once we have 3rd party code.

Reply:
I think you are assuming special knowledge about some properties. What if there are two properties that have stack?
I'd rather attribute it correctly, and give host very simple and clear message - here are tags and here are how to think about them, Anything implicit like stack here asks for trouble.

Reply:
I'll reconsider this in https://github.com/microsoft/FluidFramework/issues/5559